File: 0071-simplify-path/solution.swift
class Solution {
    func simplifyPath(_ path: String) -> String {
        var str = path.split(separator: "/")
        var stack = [String]()
        var ans = ""

        for st in str {
            if st == "." {
                // ignore
                continue
            } else if st == ".." {
                if !stack.isEmpty {
                    stack.removeLast()
                }
            } else {
                stack.append(String(st))
            }
        }

        for val in stack {
            ans += "/" + val
        }
        return ans.isEmpty ? "/" : ans
    }
}

File: 0071-simplify-path/solution.java
class Solution {
    public String simplifyPath(String path) {
        int n = path.length();
        Stack<String> st = new Stack<>();
        String[] str = path.split("/");
        String ans = "";

        for(String dir: str) {
            if(dir.equals("..")) {
                if(!st.isEmpty())
                    st.pop();
            } else if(dir.equals(".") || dir.equals("")) {
                //ignore
                continue;
            } else {
                st.push(dir);
            }
        }
        for(String val: st) {
            ans += "/" + val;
        }
        return ans.isEmpty() ? "/" : ans;
    }
}
